src/main/rx/flysky.c

```diff
@@ -0,0 +1,496 @@
+        rxRuntimeConfig->channelCount = FLYSKY_2A_CHANNEL_COUNT;
+        timings = &flySky2ATimings;
+        rxId = U_ID_0 ^ U_ID_1 ^ U_ID_2;
+        startRxChannel = flySky2ABindChannels[0];
+        A7105Init(0x5475c52A);
+        A7105Config(flySky2ARegs, sizeof(flySky2ARegs));
+    }
+    else
+    {
+        rxRuntimeConfig->channelCount = FLYSKY_CHANNEL_COUNT;
+        timings = &flySkyTimings;
+        startRxChannel = 0;
+        A7105Init(0x5475c52A);
+        A7105Config(flySkyRegs, sizeof(flySkyRegs));
+    }
+
+    if ( !IORead(bindIO) || flySkyConfig()->txId == 0)
+    {
+        binded = false;
+    }
+    else
+    {
+        binded = true;
+        txId = flySkyConfig()->txId;                                   /* load TXID        */
+        memcpy (rfChannelMap, flySkyConfig()->rfChannelMap, NUMFREQ);  /* load channel map */
+        startRxChannel = getNextChannel(0);
+    }
+
+    A7105WriteReg(A7105_0F_CHANNEL, startRxChannel);
+    A7105Strobe(A7105_RX);                                              /* start listening */
+
+    resetTimeout(micros());
+}
+
+
+void flySkySetRcDataFromPayload(uint16_t *rcData, const uint8_t *payload)
+{
+    if (rcData && payload)
+    {
+        uint32_t channelCount;
+        channelCount = (protocol == RX_SPI_A7105_FLYSKY_2A) ? (FLYSKY_2A_CHANNEL_COUNT) : (FLYSKY_CHANNEL_COUNT);
+        for (uint8_t i = 0; i < channelCount; i++)
+        {
+            rcData[i] = payload[2 * i + 1] << 8 | payload [2 * i + 0];
+        }
+    }
+}
+
+
+rx_spi_received_e flySkyDataReceived (uint8_t *payload)
+{
+    rx_spi_received_e result = RX_SPI_RECEIVED_NONE;
+    uint32_t timeStamp;
+
+    if (A7105RxTxFinished(&timeStamp))
+    {
+        uint8_t modeReg = A7105ReadReg(A7105_00_MODE);
+
+        if (((modeReg & A7105_MODE_TRSR) != 0) && ((modeReg & A7105_MODE_TRER) == 0))   /* TX complete */
+        {
+            if (binded) { A7105WriteReg(A7105_0F_CHANNEL, getNextChannel(1)); }
+            A7105Strobe(A7105_RX);
+        }
+        else if ((modeReg & (A7105_MODE_CRCF|A7105_MODE_TRER)) == 0)                    /* RX complete, CRC pass */
+        {
+            if (protocol == RX_SPI_A7105_FLYSKY_2A)
+            {
+                result = flySky2AReadAndProcess(payload, timeStamp);
+            }
+            else
+            {
+                result = flySkyReadAndProcess(payload, timeStamp);
+            }
+        }
+        else
+        {
+            A7105Strobe(A7105_RX);
+        }
+    }
+
+    if (binded) { checkTimeout(); }
+
+    if (result == RX_SPI_RECEIVED_BIND)
+    {
+        flySkyConfigMutable()->txId = txId;                                 /* store TXID */
+        memcpy (flySkyConfigMutable()->rfChannelMap, rfChannelMap, NUMFREQ);/* store channel map */
+        flySkyConfigMutable()->protocol = protocol;
+        writeEEPROM();
+    }
+
+    return result;
+}
+
+
+static inline rx_spi_received_e flySky2AReadAndProcess (uint8_t *payload, const uint32_t timeStamp)

```

Comment:
Probably not necessary to use `inline` here, the compiler will optimise it if it's a viable optimisation.